### PR TITLE
destination-dev-null: merge cloud and OSS

### PR DESCRIPTION
**Pull request by stephane-airbyte**
## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by stephane-airbyte** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/45651?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#45651** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/45651?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#45650** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/45650?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#45649** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/45649?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#45648** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/45648?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stephane-airbyte and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-e2e-test/metadata.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ data:
   connectorSubtype: unknown
   connectorType: destination
   definitionId: 2eb65e87-983a-4fd7-b3e3-9d9dc6eb8537
```
Comment (johnny-schmidt) [inline airbyte-integrations/connectors/destination-e2e-test/metadata.yaml:4]: Just to clarify, if we drop the cloud version but don't change the definitionId or name of the oss version, then we effectively create a new cloud connector with a new name? That is what we want? All existing connections using dev-null will just be parked at the old version until they switch over?

File: airbyte-integrations/connectors/destination-dev-null/src/main/kotlin/io/airbyte/integrations/destination/dev_null/TestingDestinations.kt [hunk 2]
```diff
@@ -42,6 +48,33 @@ constructor(
...
     }
 
+    override fun spec(): ConnectorSpecification {
+        if (isCloudDeployment) {
+            return super.spec()
+        } else {
+            /** 1. Update the title. 2. Only keep the "silent" mode. */
+            val spec = super.spec()
+
+            (spec.connectionSpecification as ObjectNode).put("title", DEV_NULL_DESTINATION_TITLE)
```
Comment (johnny-schmidt) [inline airbyte-integrations/connectors/destination-dev-null/src/main/kotlin/io/airbyte/integrations/destination/dev_null/TestingDestinations.kt:58]: This is the cloud title also?

Comment (stephane-airbyte) [reply airbyte-integrations/connectors/destination-dev-null/src/main/kotlin/io/airbyte/integrations/destination/dev_null/TestingDestinations.kt:58]: I guess I fucked that one up? I reverted them. I'll fix

Comment (stephane-airbyte) [reply airbyte-integrations/connectors/destination-dev-null/src/main/kotlin/io/airbyte/integrations/destination/dev_null/TestingDestinations.kt:58]: actually, why should we use different names, now that they're the same connector?

Comment (johnny-schmidt) [reply airbyte-integrations/connectors/destination-dev-null/src/main/kotlin/io/airbyte/integrations/destination/dev_null/TestingDestinations.kt:58]: 🤷 Minimal disruption? Keeping parity with what exists?

**Review by johnny-schmidt — APPROVED** [review]

**Comment by stephane-airbyte** [comment]
### Merge activity

* **Sep 20, 6:38 PM EDT**: @stephane-airbyte merged this pull request with [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/45651).

**Pull request merged**